src/components/calcite-dropdown-item/calcite-dropdown-item.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State
+} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+import DropdownInterface from "../../interfaces/DropdownInterface";
+
+@Component({
+  tag: "calcite-dropdown-item",
+  styleUrl: "calcite-dropdown-item.scss",
+  shadow: true
+})
+export class CalciteDropdownItem {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  @Prop({ reflect: true, mutable: true }) active: boolean = false;
+
+  /** @internal */
+  @Prop() currentDropdownGroup: string = this.el.parentElement.id;
```

Comment:
Coming from React, I would say an "internal prop" is just state. Would `@State()` be appropriate here?

Reply:
Good catch